napari/_qt/qt_main_window.py

```diff
@@ -31,12 +33,35 @@
     QStatusBar,
     QVBoxLayout,
 )
-from qtpy.QtCore import Qt  # noqa: E402
+from qtpy.QtCore import Qt, QTimer  # noqa: E402
 from qtpy.QtGui import QKeySequence, QIcon  # noqa: E402
 from .utils import QImg2array  # noqa: E402
 from ..utils.theme import template  # noqa: E402
 
 
+class NapariMainWindow(QMainWindow):
```

Comment:
can definitely discuss this, just wanted access to the main_window `closeEvent` which is easier with a subclass.

Reply:
I'm ok with this